File: rently-backend/src/main/java/com/rently/rentlybackend/dto/response/BookingResponse.java
package com.rently.rentlybackend.dto.response;

import java.math.BigDecimal;
import java.math.BigInteger;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

public record BookingResponse(
        String id,
        String propertyId,
        LocalDate checkIn,
        LocalDate checkOut,
        LocalDateTime createdAt,
        BigDecimal finalPrice,
        List<Guest> guests
) {
    public record Guest(
            String firstName,
            String lastName
    ) {}
}

File: rently-frontend/src/types/BookingResponse.ts
export type BookingResponse = {
  id: string;
  propertyId: string;
  checkIn: string;
  checkOut: string;
  createdAt: string;
  finalPrice: number;
  guests: GuestResponse[];
}

export type GuestResponse = {
    firstName: string;
    lastName: string;
}